File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        max_length = 0
        l = 0
        for r in range(len(nums)):
            if nums[r] == 0:
                k -= 1
            while k < 0:
                if nums[l] == 0:
                    k += 1
                l += 1

            max_length = max(max_length, r - l + 1)

        return max_length

File: 1046-max-consecutive-ones-iii/1046-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int n = nums.size();
        int max_w = 0, z = 0, l =0;

        for (int r = 0; r < n; r++)
        {
            if (nums[r] == 0)
                z++;

            while (z > k)
            {
                if (nums[l] == 0)
                    z--;
                l++;
            }
            int w = r - l + 1;
            max_w = max(max_w, w);
        }
        return max_w;
    }
};